### PR TITLE
feat: improve code pattern to allow the author to use supporting script/style files

### LINKED ISSUE #9522: `CodePattern` component should allow for preloading supporting code
(See the [Google doc](https://docs.google.com/document/d/1Vzc1ieHE8lZqWOpXVN9v9oaahfrYwAPYZ6MaeyeaoN8/edit?usp=sharing&resourcekey=0-O7BMDa07Pj6b894Y5Eag8w) for the full background.)

Inspired by CodePen, we propose an improvement to the existing [CodePattern component](https://github.com/GoogleChrome/web.dev/blob/main/src/site/_includes/components/CodePattern.js) in web.dev, namely the possibility of preloading the code pattern with external CSS, JS, and potentially HTML resources.

Unlike with CodePen, though, were such preloads are not easily exposed to the user, we propose to initially hide preloads via a [filter](https://github.com/GoogleChrome/web.dev/blob/f3d6e739f9cb0351e064b8506c1cdbfd6a36e084/src/lib/patterns.js#L64-L67), but add expandable tabs to the component that teases initially shrunk supporting resources under a More… tab, which the user can then click to see the expanded tabs, like Supporting JS in the mock below. 

Screenshot 2023-02-07 at 13 22 44

### PR DESCRIPTION
**Pull request by mamieorine**
Fixes #9522 

Changes proposed in this pull request:

- Improve code pattern to allow the author to add the supporting files (CSS and JS)
- Add `limit` attribute into Tabs component to limit the number of tabs showing initially
- Add expendable tab, named **MORE...**

### CODE REVIEW

**Review by jimjiminyjimjim — COMMENTED** [review]
LGTM

File: src/lib/components/Tabs/index.js [hunk 1]
```diff
@@ -64,6 +71,27 @@ export class Tabs extends BaseElement {
     `;
   }
 
+  loadmoreTab() {
```
Comment (matthiasrohmer) [inline src/lib/components/Tabs/index.js:74]: Nit, casing: `loadMoreTab`, also applies to `loadMorePanel` and `onClickLoadMore`.

File: src/lib/components/Tabs/index.js [hunk 2]
```diff
@@ -117,6 +147,10 @@ export class Tabs extends BaseElement {
     `;
   }
 
+  loadmorePanel() {
+    return html`<div class="web-tabs__panel" role="tabpanel" hidden></div>`;
+  }
```
Comment (matthiasrohmer) [inline src/lib/components/Tabs/index.js:152]: I think that's fine to inline 🙂 doesn't take any arguments and is just one line.

File: src/site/_includes/components/CodePattern.js [hunk 3]
```diff
@@ -41,6 +41,11 @@ module.exports = (patternId, height) => {
       const type = prismTypes.includes(asset.type) ? asset.type : 'text';
       assetLines.push(asset.content.split('\n').length);
 
+      const isSupportingAsset = asset.name.includes('supporting');
+      const tabName = isSupportingAsset
+        ? `Supporting ${asset.type}`
```
Comment (matthiasrohmer) [inline src/site/_includes/components/CodePattern.js:46]: This would need to go through i18n.

**Review by matthiasrohmer — COMMENTED** [review]
This works, at least for the tabs part. I am not too sure if the supporting files would need to be somehow incorporated into the "built" demo though because I am lacking context. Looping in @devnook.

**Review by matthiasrohmer — COMMENTED** [review]
Synced with Ewa, despite the few style nits this LGTM. Thanks, Maya! 🙂

**Review by matthiasrohmer — APPROVED** [review]

**Pull request merged**